src/MeetingsSDKAdapter.js

```diff
@@ -216,19 +210,24 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
    * @param {string} ID ID of the meeting to mute audio
    * @memberof MeetingsSDKAdapter
    */
-  muteAudioMeeting(ID) {
+  async handleLocalAudio(ID) {
     const sdkMeeting = this.fetchMeeting(ID);
+    const audioEnabled = this.meetings[ID].localAudio.getAudioTracks()[0].enabled;
+    let eventName;
 
-    sdkMeeting
-      .muteAudio()
-      .then(() => {
-        // Due to SDK limitation around promises, we need to emit a custom event for audio mute action
-        sdkMeeting.emit(EVENT_MEDIA_LOCAL_MUTE, {control: MUTE_AUDIO_CONTROL});
-      })
-      .catch((error) => {
-        // eslint-disable-next-line no-console
-        console.error(`Unable to mute audio for meeting "${ID}"`, error);
-      });
+    try {
+      if (audioEnabled) {
+        await sdkMeeting.muteAudio();
+      } else {
+        await sdkMeeting.unmuteAudio();
+      }
+
+      // Due to SDK limitation around promises, we need to emit a custom event for audio mute action
+      sdkMeeting.emit(eventName, {control: AUDIO_CONTROL, state: !audioEnabled});
```

Comment:
Would it be better to re-assign `audioEnabled` after muting/unmuting (in the `if/else`?
This way we send the true current state of the audio track?

Reply:
why does that matter?